### DISCUSSION TITLE
Login customer programmatically in frontend

### DISCUSSION THREAD

**Discussion by abcpremium** [pimcore/pimcore#9925] [Q&A / Support]
Hey guys,

I am quite new to Pimcore and I haven't found a solution online. I have the demo package with the CustomerManagementFramework installed locally and created an own route with an empty template (just containing a login form with username and password input). I want to post this data to my LoginController and want to login the customer, then redirect him to another empty template. But how can I login a customer? I do not mean a user, but a customer. I haven't found any solution...

Thanks in advance! This is what I have so far...

```
public function authenticateAction(Request $request): Response
    {
        $req = $request->request;
        $username = $req->get('_username');
        $password = $req->get('_password');

        $customer = $this->_customer->getActiveCustomerByEmail($username);
        if ($customer === false) {
            return new Response('Customer not found');
        }

       /* Login procedure here? */
    }
```

**Comment by Humb3l** [comment]
If I understood you correctly, you want to overwrite the default login Template? 
In this case you need to overwrite the standard login template.
In Addition to that you can also overwrite the default controller.

In this discussion is described how you can do that
https://github.com/pimcore/pimcore/discussions/9897

I hope I could help you :)

**Reply by abcpremium** [reply]
Thank you for the suggestion, but no, I do not want to login users, but customers. Like in the ecommerce demo.
